src/extension/background-script/actions/liquid/generatePrivateKey.ts

```diff
@@ -0,0 +1,55 @@
+import * as secp256k1 from "@noble/secp256k1";
+import Hex from "crypto-js/enc-hex";
+import sha512 from "crypto-js/sha512";
+import type { MessagePrivateKeyGenerate } from "~/types";
+
+import state from "../../state";
+
+const generatePrivateKey = async (message: MessagePrivateKeyGenerate) => {
+  const type = message?.args?.type;
+
+  const privateKey =
+    type === "random" ? generateRandomKey() : await deriveKey();
+
+  if (privateKey)
+    return {
+      data: {
+        privateKey: secp256k1.utils.bytesToHex(privateKey),
+      },
+    };
+  else
+    return {
+      error: "Error generating private key.",
+    };
+};
+
+const generateRandomKey = () => {
+  return secp256k1.utils.randomPrivateKey();
+};
+
+const deriveKey = async () => {
+  const connector = await state.getState().getConnector();
+  try {
+    const response = await connector.signMessage({
```

Comment:
Do we want to do the same derivation for liquid keys? 🤔 Unsure whether we should just generate a new set of keys. The derivation method allows you to exactly generate `1` key, if this key will ever get compromised you will have to generate a new key and backup it anyway. 

Does the advantage of not having to backup this key separately justify this additional complexity? (not necessarily from the technical perspective, but also from the user perspective, having to explain the different options, etc)

Reply:
not sure what you mean here.